src/bindings/python/PyTransform.cpp

```diff
@@ -1,310 +1,41 @@
-        self->isconst = true;
-        std::string message = "Base Transforms class can not be instantiated.";
-        PyErr_SetString( PyExc_RuntimeError, message.c_str() );
-        return -1;
-    }
-
-    void PyOCIO_Transform_delete(PyOCIO_Transform *self)
-    {
-        DeletePyObject<PyOCIO_Transform>(self);
-    }
-
-    PyObject * PyOCIO_Transform_str(PyObject * self)
-    {
-        OCIO_PYTRY_ENTER()
-        ConstTransformRcPtr transform = GetConstTransform(self, true);
-        std::ostringstream out;
-        out << *transform;
-        return PyString_FromString(out.str().c_str());
-        OCIO_PYTRY_EXIT(NULL)
-    }
-
-    PyObject * PyOCIO_Transform_isEditable(PyObject * self, PyObject *)
-    {
-        return PyBool_FromLong(IsPyTransformEditable(self));
-    }
-
-    PyObject * PyOCIO_Transform_createEditableCopy(PyObject * self, PyObject *)
-    {
-        OCIO_PYTRY_ENTER()
-        ConstTransformRcPtr transform = GetConstTransform(self, true);
-        TransformRcPtr copy = transform->createEditableCopy();
-        PyOCIO_Transform * pycopy = PyTransform_New(copy);
-        pycopy->constcppobj = new ConstTransformRcPtr();
-        pycopy->cppobj = new TransformRcPtr();
-        *pycopy->cppobj = copy;
-        pycopy->isconst = false;
-        return (PyObject *) pycopy;
-        OCIO_PYTRY_EXIT(NULL)
-    }
-
-    PyObject * PyOCIO_Transform_getDirection(PyObject * self, PyObject *)
-    {
-        OCIO_PYTRY_ENTER()
-        ConstTransformRcPtr transform = GetConstTransform(self, true);
-        TransformDirection dir = transform->getDirection();
-        return PyString_FromString(TransformDirectionToString(dir));
-        OCIO_PYTRY_EXIT(NULL)
-    }
-
-    PyObject * PyOCIO_Transform_setDirection(PyObject * self, PyObject * args)
-    {
-        OCIO_PYTRY_ENTER()
-        TransformDirection dir;
-        if (!PyArg_ParseTuple(args, "O&:setDirection",
-            ConvertPyObjectToTransformDirection, &dir)) return NULL;
-        TransformRcPtr transform = GetEditableTransform(self);
-        transform->setDirection(dir);
-        Py_RETURN_NONE;
-        OCIO_PYTRY_EXIT(NULL)
-    }
 
+void bindPyTransform(py::module & m)
+{ 
+    auto cls = py::class_<Transform, TransformRcPtr /* holder */>(m, "Transform")
+        .def("validate", &Transform::validate)
+        .def("getDirection", &Transform::getDirection)
+        .def("setDirection", &Transform::setDirection, "dir"_a);
+
+    defStr(cls);
+
+    bindPyFormatMetadata(m);
+    bindPyDynamicProperty(m);
+
+    // Subclasses
+    bindPyAllocationTransform(m);
+    bindPyCDLTransform(m);
+    bindPyColorSpaceTransform(m);
+    bindPyDisplayTransform(m);
+    bindPyExponentTransform(m);
+    bindPyExponentWithLinearTransform(m);
+    bindPyExposureContrastTransform(m);
+    bindPyFileTransform(m);
+    bindPyFixedFunctionTransform(m);
+    bindPyGroupTransform(m);
+    bindPyLogAffineTransform(m);
+    bindPyLogTransform(m);
+    bindPyLookTransform(m);
```

Comment:
Wow, totally missed it! Thanks for catching that. I think I probably passed right over it since it's interface is very similar to LogAffineTransform. Added.